extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1468,3 +1301,46 @@ export class AmpA4A extends AMP.BaseElement {
    */
   emitLifecycleEvent(unusedEventName, opt_extraVariables) {}
 }
+
+/**
+ * Decode the `AMP-Fast-Fetch-Signature` header, in the format
+ * `{signingServiceName}:{keypairId}:{base64Signature}`, to a signature info
+ * object.
+ *
+ * @param {?string} headerValue
+ * @return {?SignatureInfoDef}
+ */
+export function decodeSignatureHeader(headerValue) {
+  if (headerValue) {
+    const match =
+        new RegExp(
+            '^([A-Za-z0-9._-]+):([A-Za-z0-9._-]+):' +
+            '((?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}[A-Za-z0-9+/=]=)?)$')
+            .exec(headerValue);
+    if (match) {
+      return /** @type {?SignatureInfoDef} */ ({
+        signingServiceName: match[1],
+        keypairId: match[2],
+        signature: base64DecodeToBytes(match[3]),
+      });
+    }
+  }
+  return null;
+}
+
+/**
+ * Decode the `X-Creativesize` header, in the format `{width}x{height}`, to a
+ * size info object.
+ *
+ * @param {?string} headerValue
+ * @return {?SizeInfoDef}
+ */
+export function decodeSizeHeader(headerValue) {
+  if (headerValue) {
+    dev().assert(new RegExp('[0-9]+x[0-9]+').test(headerValue));
```

Comment:
Should we guard/report in production code when we have an invalid header value?  Should there be another assert that width/height are not NaN?